application-forum-ui/src/main/resources/ForumCode/Actions.xml

```diff
@@ -65,26 +67,34 @@
         $response.setStatus(403)
       #end
     #end
-  #elseif("$!request.action" == "addtopic")
+  #elseif("$!request.action" == 'addtopic')
     #if("$!request.title.trim()" != '')
       #if("$!request.space" != '')
-        #set($newTopicName = $request.page)
         #set($topicTitle = "$!request.title.trim()")
-        #set($topicDoc = $xwiki.getDocument($newTopicName))
+        #set($topicSpace = "$!request.space")
+        #if($supportsNS)
+          #set($topicSpaceReference = $services.model.resolveSpace($topicSpace))
+          #set($forum = $topicSpaceReference.extractReference('SPACE').name)
+          #set($forumsRoot = $services.model.serialize($topicSpaceReference.parent))
+          #set($newDocReference = $services.model.createDocumentReference($doc.wiki, [$forumsRoot, $forum, $topicTitle], 'WebHome'))
```

Comment:
```
#set ($topicSpaceReference = $services.model.createSpaceReference($topicTitle, $forumSpaceReference))
#set ($newDocReference = $services.model.createDocumentReference('WebHome', $topicSpaceReference))
```
The above format makes no assumption about the forum app's (forums root) location, i.e. if it`s a top level space or not, so it's cleaner on the long run.